### PR TITLE
port from lib

### PR DESCRIPTION
**Pull request by jrit**
@Tathanen pulled this out of lib as part of my effort to gut that thing. AFAIK I'm the only one that ever used this as I only added it a few months ago.

### CODE REVIEW

File: src/clearable.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+angular.module( "clearable", [] )
+
+.directive( "clearable", [ "$compile", function ( $compile )
+    {
+        "use strict";
+
```
Comment (Tathanen) [inline src/clearable.js:6]: Maybe throw a "usage" comment in here, ala this: https://github.com/Tathanen/vokal-ng-lib/blob/master/directives/vokal_date_picker.js

Comment (Tathanen) [inline src/clearable.js:6]: ..or just a readme that actually says something useful instead.

**Comment by jrit** [comment]
updated readme

File: README.md [hunk 2]
```diff
@@ -1,2 +1,9 @@
 # angular-clearable
-Directive for adding a clear affordance to input fields
+
+Angular directive for adding a clear affordance to input fields
+
+## Use
+
+`<input type="text" data-clearable>`
+
+This will create an icon on the right of the input field that when selected clears the input field. The icon used depends on fontawesome.
```
Comment (Tathanen) [inline README.md:9]: Is it really worth including fontawesome for this one icon?  It's just an X, you could draw one with two CSS rectangles pretty trivially.

Comment (jrit) [inline README.md:9]: or use `&times;`

Comment (Tathanen) [inline README.md:9]: Well that's not very FANCY.

**Comment by jrit** [comment]
I'm going to merge and open an issue to remove fontawesome as a dep, same on other PR

**Pull request merged**